plugin/storage/es/spanstore/reader_test.go

```diff
@@ -0,0 +1,420 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package spanstore
+
+import (
+	"testing"
+
+	"github.com/olivere/elastic"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"encoding/json"
+	"errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/uber/jaeger/model"
+	"github.com/uber/jaeger/pkg/es/mocks"
+	"github.com/uber/jaeger/pkg/testutils"
+	"github.com/uber/jaeger/storage/spanstore"
+	"time"
+)
+
+type spanReaderTest struct {
+	client    *mocks.Client
+	logger    *zap.Logger
+	logBuffer *testutils.Buffer
+	reader    *SpanReader
+}
+
+func withSpanReader(fn func(r *spanReaderTest)) {
+	client := &mocks.Client{}
+	logger, logBuffer := testutils.NewLogger()
+	r := &spanReaderTest{
+		client:    client,
+		logger:    logger,
+		logBuffer: logBuffer,
+		reader:    NewSpanReader(client, logger),
+	}
+	fn(r)
+}
+
+func TestNewSpanReader(t *testing.T) {
+	withSpanReader(func(r *spanReaderTest) {
+		var reader spanstore.Reader = r.reader // check API conformance
+		assert.NotNil(t, reader)
+	})
+}
+
+func TestSpanReader_GetTrace(t *testing.T) {
+	// TODO: write test once done with function
+	// currently not doing anything, only for code coverage, ignore for code review
+	withSpanReader(func(r *spanReaderTest) {
+		s, e := r.reader.GetTrace(model.TraceID{})
+		assert.Nil(t, s)
+		assert.Nil(t, e)
+	})
+}
+
+func TestSpanReader_findIndicesEmptyQuery(t *testing.T) {
+	withSpanReader(func(r *spanReaderTest) {
+		existsService := &mocks.IndicesExistsService{}
+		existsService.On("Do", mock.AnythingOfType("*context.emptyCtx")).Return(true, nil)
+		r.client.On("IndexExists", mock.AnythingOfType("string")).Return(existsService)
+
+		actual := r.reader.findIndices(spanstore.TraceQueryParameters{})
+
+		today := time.Now().Format("2006-01-02")
+		yesterday := time.Now().AddDate(0, 0, -1).Format("2006-01-02")
+		twoDaysAgo := time.Now().AddDate(0, 0, -2).Format("2006-01-02")
+
+		expected := []string{
+			"jaeger-" + today,
```

Comment:
also, the logic for creating an index name for a given Time should be encapsulated into a reusable function, not spread out.